sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -2125,6 +2266,65 @@ node nodeA = myGraph.add([&](handler& cgh) {
 dynParamAccessor.update(bufferB.get_access());
 ----
 
+=== Dynamic Event Update
+
+Example which shows how to create a dependency between two graphs using dynamic
+events which are then updated between executions.
+
+[source,c++]
+----
+...
+
+using namespace sycl;
+namespace sycl_ext = sycl::ext::oneapi::experimental;
+
+queue myQueue;
+auto myContext = myQueue.get_context();
+auto myDevice = myQueue.get_device();
+
+// Create the first graph
+sycl_ext::command_graph graphA(myContext, myDevice);
+
+// Add a node to graphA
+sycl_ext::node nodeA = graphA.add(...);
+
+sycl_ext::command_graph<sycl_ext::graph_state::executable> execGraphA = 
+  graphA.finalize();
+
+// Get an event for the next execution of nodeA in execGraphA
+event eventA = execGraph.get_event(nodeA);
+
+// Create the second graph
+sycl_ext::command_graph graphB(myContext, myDevice);
+
+// Create the dynamic event which will wrap eventA and use it as a dependency
+// for a node in graphB
+sycl_ext::dynamic_event dynEvent{eventA};
+
+auto nodeB = graphB.add([&](handler& CGH){
+  CGH.depends_on(dynEvent);
+  CGH.parallel_for(...);
+});
+
+auto execGraphB = graphB.finalize({sycl_ext::property::graph::updatable});
+
+// Submit both graphs for execution
+myQueue.submit(execGraphA);
+myQueue.submit(execGraphB);
+myQueue.wait_and_throw();
+
+// Update the dynamic event with a new event frome execGraphA to reflect the
+// next execution
+dynEvent.update(execGraphA.get_event(nodeA));
+// Update execGraphB with the affected node to reflect those changes
+execGraphB.update(nodeB);
```

Comment:
> In particular, perhaps there is a way to leverage the fact that native CUDA events are auto-resetting, and make dynamic_event somehow smart enough to make the graph update a no-op when the same CUDA event can be reused?

I think it is probably possible, but as I mentioned in another comment I think there's quite a high implementation burden for that. Since we both have to map to multiple, quite different APIs on the UR level, and both the graph runtime and UR implementations are very much not set up to support that kind of information sharing between graphs/command-buffers.

I think because of that burden if we were going to do that it would need to be done as a separate piece of work, so I would be in favour of leaving this design as is for now. It should be easy to update the spec to a new behaviour in the future, if we can commit to the implementation in the future.

Reply:
> Graphs are primarily used for latency-sensitive workloads, right? And here we mandate that if we want to use dynamic events as a dependency, we must update the graph every time it's submitted.

my comment from user perspective, we always like the static graph without change, but if we are unable to keep the graph always the same (due to use case), we would have to use the dynamic (parameters or events) to update the graph, such overhead should be less than recording/creating the graph again.

Reply:
> I would argue that it's a concern for many use-cases. Graphs are primarily used for latency-sensitive workloads, right? And here we mandate that if we want to use dynamic events as a dependency, we must update the graph every time it's submitted.

I was referring to your use-case in contrast to the use-case example that I gave which may not use these kind of dependencies between graphs, and thus would not need to use update for their use-case, rather than trying to imply that your use-case is the only one with such concerns.

I agree that this is a concern for most and is the primary driver behind using graphs. Ideally having to update, especially a small number of nodes/events, will not outweigh the gains in reduced host latency from using graphs so there would still be a net performance increase. The effectiveness of graphs as a whole is workload dependent though so this may not be true for all cases.